lib/Optimizer/Builder/Factory.cpp

```diff
@@ -225,19 +225,27 @@ bool factory::hasHiddenSRet(FunctionType funcTy) {
   auto numResults = funcTy.getNumResults();
   return numResults > 1 ||
          (numResults == 1 && funcTy.getResult(0)
-                                 .isa<cc::StdvecType, cc::StructType,
+                                 .isa<cc::SpanLikeType, cc::StructType,
                                       cc::ArrayType, cc::CallableType>());
 }
 
+cc::StructType factory::stlStringType(MLIRContext *ctx) {
+  auto i8Ty = IntegerType::get(ctx, 8);
+  auto ptrI8Ty = cc::PointerType::get(i8Ty);
+  auto i64Ty = IntegerType::get(ctx, 64);
+  auto padTy = cc::ArrayType::get(ctx, i8Ty, 16);
```

Comment:
Would it make sense to add a `static_assert` here to validate our layout assumption?
e.g.,
```
static_assert(sizeof(std::string) == sizeof(char*) + sizeof(std::size_t) + 16);
```